autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/FlowCommunicator.java

```diff
@@ -0,0 +1,51 @@
+package com.sequenceiq.periscope.monitor.handler;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static java.util.stream.Collectors.toMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.Lists;
+import com.sequenceiq.cloudbreak.client.CloudbreakInternalCrnClient;
+import com.sequenceiq.flow.api.model.FlowCheckResponse;
+
+@Service
+public class FlowCommunicator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FlowCommunicator.class);
+
+    private static final Integer PAGE_SIZE = 50;
+
+    private static final Integer PAGE = 0;
+
+    @Inject
+    private CloudbreakInternalCrnClient internalCrnClient;
+
+    public Map<Long, FlowCheckResponse> getFlowStatusFromFlowIds(Map<String, Long> activityIdsByFlowIds) {
+        LOGGER.info("Sending request to fetch FlowStatus of {} flowIds", activityIdsByFlowIds.size());
+        List<String> allFlowIds = newArrayList(activityIdsByFlowIds.keySet());
+        List<FlowCheckResponse> flowCheckResponses = newArrayList();
+        Lists.partition(allFlowIds, PAGE_SIZE).parallelStream().forEach(flowIds ->
```

Comment:
Switch to a sequential stream?
And apply a total limit on how many events we will request in a single invocation of ActivityUpdateEvaluator (Avoid bombarding CB with a large number of requests, if we end up with a backlog).

Also: worth adding a metric here, to track this backlog.